sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/utils/SDKUtil.java

```diff
@@ -79,4 +98,75 @@ public static ArrayList listFiles(String sourceImageFolder,
     return (Object[]) input[i];
   }
 
+  public static List<CarbonFile> extractFilesFromFolder(String path,
+      String suf, Configuration hadoopConf) {
+    List dataFiles = listFiles(path, suf, hadoopConf);
+    List<CarbonFile> carbonFiles = new ArrayList<>();
+    for (Object dataFile: dataFiles) {
+      carbonFiles.add(FileFactory.getCarbonFile(dataFile.toString(), hadoopConf));
+    }
+    if (CollectionUtils.isEmpty(dataFiles)) {
+      throw new RuntimeException("No file found at given location. Please provide" +
+          "the correct folder location.");
+    }
+    return carbonFiles;
+  }
+
+  public static DataFileStream<GenericData.Record> buildAvroReader(CarbonFile carbonFile,
+       Configuration configuration) throws IOException {
+    try {
+      GenericDatumReader<GenericData.Record> genericDatumReader =
+          new GenericDatumReader<>();
+      DataFileStream<GenericData.Record> avroReader =
+          new DataFileStream<>(FileFactory.getDataInputStream(carbonFile.getPath(),
+          -1, configuration), genericDatumReader);
+      return avroReader;
+    } catch (FileNotFoundException ex) {
+      throw new FileNotFoundException("File " + carbonFile.getPath()
+          + " not found to build carbon writer.");
+    } catch (IOException ex) {
+      if (ex.getMessage().contains("Not a data file")) {
+        throw new RuntimeException("File " + carbonFile.getPath() + " is not in avro format.");
+      } else {
+        throw ex;
+      }
+    }
+  }
+
+  public static Reader buildOrcReader(String path, Configuration conf) throws IOException {
+    try {
+      Reader orcReader = OrcFile.createReader(new Path(path),
+          OrcFile.readerOptions(conf));
+      return orcReader;
+    } catch (FileFormatException ex) {
+      throw new RuntimeException("File " + path + " is not in ORC format");
+    } catch (FileNotFoundException ex) {
+      throw new FileNotFoundException("File " + path + " not found to build carbon writer.");
+    }
+  }
+
+  public static ParquetReader<GenericRecord> buildPqrquetReader(String path, Configuration conf)
```

Comment:
Please correct spelling mistake for parquet in method name.

Reply:
Done